src/include/DistributedDomain.hpp

```diff
@@ -685,20 +685,21 @@ class DistributedDomain
     }
 
     template <class Dataset>
-    void updateSmoothingLength(Dataset &d)
+    void updateSmoothingLengthImpl(Task &t, Dataset &d)
     {
         const T c0 = 7.0;
         const T exp = 1.0 / 3.0;
-
-        const size_t n = clist.size();
-        const int *neighborsCount = d.neighborsCount.data();
-        const int ng0 = d.ng0;
+        
+        const int ng0 = Task::ng0;
+        const int *neighborsCount = t.neighborsCount.data();
         T *h = d.h.data();
 
-#pragma omp parallel for
-        for (int pi = 0; pi < n; pi++)
+        size_t  n = t.clist.size();
+
+#pragma omp parallel for schedule(guided)
+        for (size_t pi = 0; pi < n; pi++)
         {
-            const int i = clist[pi];
+            int i = t.clist[pi];
```

Comment:
Some things could be const here. The i, n and I think also the Task &t function parameter. If Task &t can be const, then task variable in for loop in line 716 can be const as well